File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # freq map to keep track of what we have seen 
        f={}

        for i in range(len(nums)):
            #if we alr have the second number we require in f
            diff=target-nums[i]
            if diff in f:
                return [f[diff],i]
            else:
                f[nums[i]]=i
        return []

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {

        std::unordered_map<int,int> freq;
        freq[0]=1;
        int sum=0;
        int ans=0;
        for (int n:nums){
            sum+=n;
            ans+=freq[sum-k];
            freq[sum]++;
        }
        return ans;
    }
};